core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsAddressCache.java

```diff
@@ -15,32 +13,22 @@
  * @since 7.0
  */
 public class JGroupsAddressCache {
-   private static final ConcurrentMap<Address, JGroupsAddress> addressCache =
-         new ConcurrentHashMap<>();
+   private static final Map<ExtendedUUID, JGroupsAddress> addressCache = new ConcurrentHashMap<>();
 
-   public static org.infinispan.remoting.transport.Address fromJGroupsAddress(Address jgroupsAddress) {
+   public static JGroupsAddress fromExtendedUUID(ExtendedUUID addr) {
       // New entries are rarely added after startup, but computeIfAbsent synchronizes every time
-      JGroupsAddress ispnAddress = addressCache.get(jgroupsAddress);
-      if (ispnAddress != null) {
-         return ispnAddress;
-      }
-      return addressCache.computeIfAbsent(jgroupsAddress, ignore -> {
-         if (jgroupsAddress instanceof ExtendedUUID) {
-            return new JGroupsAddress((ExtendedUUID) jgroupsAddress);
-         } else if (jgroupsAddress instanceof UUID uuid) {
-            return new JGroupsAddress(new ExtendedUUID(uuid.getMostSignificantBits(), uuid.getLeastSignificantBits()));
-         } else {
-            throw new IllegalStateException("Unexpected address type: " + jgroupsAddress.getClass().getName());
-         }
-      });
+      var existing = addressCache.get(addr);
+      return existing == null ?
```

Comment:
Can't we simplify this to just `return addressCache.computeIfAbsent(addr, JGroupsAddress::new);`?

Reply:
Happy to keep as is, I missed the comment above :man_facepalming: